File: 1633-minimum-number-of-increments-on-subarrays-to-form-a-target-array/1633-minimum-number-of-increments-on-subarrays-to-form-a-target-array.cpp
class Solution {
public:
    int minNumberOperations(vector<int>& target) {
        int n = target.size();
        int ans = target[0];

        for(int i = 1; i < n; ++i){
            int diff = target[i] - target[i - 1];
            if(diff > 0) ans += diff;
        }
        
        return ans;
    }
};

File: 1633-minimum-number-of-increments-on-subarrays-to-form-a-target-array/1633-minimum-number-of-increments-on-subarrays-to-form-a-target-array.erl
-spec min_number_operations(Target :: [integer()]) -> integer().
min_number_operations(Target) ->
    helper([0 | Target]).
    
helper([]) -> 0;
helper([A, B | T]) when B > A -> (B - A) + helper([B | T]);
helper([H | T]) -> helper(T).